### PR TITLE
Merge coverage

### LINKED ISSUE #4494: combine `cypress` and `vitest` coverage reports
Currently, Mermaid uses only `vitest` to report coverage, and `vitest` doesn't run `renderer`, `svgDraw`, etc. files, which shows the wrong results (57%):

[Coverage Status](https://coveralls.io/github/mermaid-js/mermaid?branch=master)
 
The main purpose of coverage is to know what have been test and highlight possible edge cases, but with these misleading reports, it would be hard.
 
Suggest solutions:
 
1. Create a script that takes reports from `vitest` and `cypress` coverages and then combines them.
 
Here is an [example](https://rafaelalmeidatk.com/blog/merging-coverage-reports-from-jest-and-cypress) for someone who combines `jest` and `cypress` coverage reports.
 
2. Make `vitest` handle both headless and unit testing; that will show better coverage, and from [vitest.dev](https://vitest.dev/guide/comparisons.html#cypress):
 
    > We suggest using Vitest for all headless logic in your application and Cypress for all browser-based logic.

    And:

    > We believe that Cypress isn't a good option for unit testing headless code, but that using Cypress (for E2E and component testing) and Vitest (for unit tests) would cover your app's testing needs.

3. Migrate to a new tool for both types of testing.
 
4. Migrate to the new visual testing tool for faster reports.

What do you think? Do you have something else in mind?
 
IMO, the first solution is good, but it will take more time to report, so the second looks like a better solution.

### PR DESCRIPTION
**Pull request by sidharthv96**
## :bookmark_tabs: Summary

Merge vitest and cypress coverages.

Resolves #4494

## :straight_ruler: Design Decisions

Adopted from https://rafaelalmeidatk.com/blog/merging-coverage-reports-from-jest-and-cypress

### :clipboard: Tasks

Make sure you

- [ ] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [ ] :computer: have added unit/e2e tests (if appropriate)
- [ ] :notebook: have added documentation (if appropriate)
- [ ] :bookmark: targeted `develop` branch

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +31,8 @@
     "cypress:open": "cypress open",
     "e2e": "start-server-and-test dev http://localhost:9000/ cypress",
     "e2e:coverage": "VITE_COVERAGE=true pnpm e2e",
+    "coverage:merge": "ts-node-esm scripts/coverage.ts",
+    "coverage": "pnpm test:coverage && pnpm e2e:coverage && pnpm coverage:merge",
```
Comment (Yokozuna59) [inline package.json:35]: The `test:coverage` script is in watch mode. The command won't be terminated.
